File: frontEnd/react-frontend/src/services/productService.js
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8080/api/products';

class ProductService {
  // Get all products
  getAllProducts() {
    return axios.get(API_BASE_URL);
  }

  // Get product by ID
  getProductById(id) {
    return axios.get(`${API_BASE_URL}/${id}`);
  }

  // Create new product
  createProduct(productData) {
    return axios.post(API_BASE_URL, productData);
  }

  // Update product
  updateProduct(id, productData) {
    return axios.put(`${API_BASE_URL}/${id}`, productData);
  }

  // Delete product
  deleteProduct(id) {
    return axios.delete(`${API_BASE_URL}/${id}`);
  }
}

export default new ProductService();

File: backend/inventory_app/src/main/java/net/java/inventory_app/service/ProductService.java
package net.java.inventory_app.service;

import net.java.inventory_app.entity.Product;
import java.util.List;

public interface ProductService {
    
    // Create a new product
    Product createProduct(Product product);
    
    // Get all products
    List<Product> getAllProducts();
    
    // Get product by ID
    Product getProductById(Long id);
    
    // Update product
    Product updateProduct(Long id, Product product);
    
    // Delete product
    void deleteProduct(Long id);
    
    // Get most bought products
    List<Product> getMostBoughtProducts();
    
    // Get low stock products
    List<Product> getLowStockProducts(Integer threshold);
}